src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -548,4 +550,31 @@ public ICitizenData getRandomCitizen()
     {
         return (ICitizenData) citizens.values().toArray()[random.nextInt(citizens.values().size())];
     }
+
+    @Override
+    public void updateCitizenSleep(final boolean sleep)
+    {
+        this.isCitizensSleeping = sleep;
+    }
+
+    @Override
+    public void onCitizenSleep()
+    {
+
+
+        for (final ICitizenData citizenData : citizens.values())
```

Comment:
This is actually more then needed. We have already a way to check on the sleep handler if a citizen is sleeping. So you only need to check this, no need for additional variables.

Reply:
`this.isCitizensSleeping` is used to mark that all citizens are sleeping, not just a particular citizen.

Reply:
I mean, that is not necessary. You can just, when a citizen lies down, check if all other citizen are sleeping already. And if so, print the message.

Reply:
Oh, I remember now. The citizen, for some reason, calls `trySleep` multiple times even after they sleep which lead to several notifications being triggered. This was my way to reset the notifications until the next day.

Reply:
Then there seems to be code missing in the PR, does this compile?